pkg/cluster/operation/action.go

```diff
@@ -591,6 +593,24 @@ func StopComponent(ctx context.Context,
 				logger.Debugf("Ignored stopping %s for %s:%d", name, ins.GetHost(), ins.GetPort())
 				continue
 			}
+		case spec.ComponentCDC:
+			nctx := checkpoint.NewContext(ctx)
+			if !forceStop {
+				// when scale-in cdc node, each node should be stopped one by one.
+				cdc, ok := ins.(spec.RollingUpdateInstance)
+				if !ok {
+					panic("cdc should support rolling upgrade, but not")
+				}
+				err := cdc.PreRestart(nctx, topo, int(timeout), tlsCfg)
```

Comment:
What's the timeout? can we configure the value?

Reply:
the timeout if `transfer-timeout`, default to `300s` in scale-in scenario.

Reply:
What about restart and stop scenario?

Reply:
For the `Restart` and the`Stop`, the timeout is not set. all instances stopped by force, so it's useless.